src/LaunchDarkly.EventSource/EventSource.cs

```diff
@@ -114,14 +125,22 @@ public EventSource(Configuration configuration)
         /// <exception cref="InvalidOperationException">The method was called after the connection <see cref="ReadyState"/> was Open or Connecting.</exception>
         public async Task StartAsync()
         {
-            var cancellationToken = _pendingRequest.Token;
-            while (!cancellationToken.IsCancellationRequested)
```

Comment:
This was the main problem.

`_pendingRequest.Token` was being used to keep track of the state of the current HTTP request - if we received bad data, we would cancel this token which would then terminate the request.

We do not want that to cause the whole stream to shut down. And we were apparently trying to avoid doing so: on [this line](https://github.com/launchdarkly/dotnet-eventsource/pull/36/files#diff-896d0f336a16ef11d2f9c3bce9244546L167), we were replacing `_pendingRequest` with a new CancellationTokenSource before cancelling the old one, in the case of a non-fatal error. But that didn't work, since the `while` loop here was still checking the _old_ `_pendingRequest`.

In any case, it's only the HTTP request that really needs a CancellationToken. For the while loop here, it makes more sense to just check explicitly whether we've been shut down.